content/push/publish.textile

```diff
@@ -0,0 +1,776 @@
+
+Ensure successful delivery, customization, and compatibility across various devices and operating systems using platform-specific requirements and best practices:
+
+
+```[json]
+{
+  "notification": {
+    "title": <string, title to display at the notification>,
+    "body": <string, text below title on the expanded notification>,
+    "icon": <string, platform-specific>,
+    "sound": <string, platform-specific>,
+    "collapseKey": <string, platform-specific, used to group notifications together>
+  },
+  "data": {
+    <key string>: <value string>,
+    ...
+  }
+}
+```
+
+Ably automatically adjusts message fields to ensure compatibility with the target push notification transport:
+
+<!--[!!Ed. note: Removed experimental component column] _=. Web (*experimental*) | @notification.title@ | @notification.body@ |@notification.icon@ |@notification.sound@ |@notification.collapseKey@ |@data@ |-->
+
+|_=. Ably field |_=. FCM |_=. APNs |
+| @notification.title@ | @notification.title@ | @aps.alert.title@ |
+| @notification.body@ | @notification.body@ | @aps.alert.body@ |
+| @notification.icon@ | @notification.icon@ | Discarded |
+| @notification.sound@ | @notification.sound@ | @aps.alert.sound@ |
+| @notification.collapseKey@ | @collapse_key@ | @aps.thread-id@ |
+| @data@ | @data@ | Merged into root object |
+
+You can use different payload formats for FCM and APNs when publishing messages to Ably. The following are examples push payloads in both formats:
+
+```[json]
+{
+  "notification": {
+    "collapseKey": "chat"
+  }
+}
+```
+
+Raw format to FCM:
+
+```[json]
+{
+  "collapse_key": "chat"
+}
+```
+
+Raw format to APNs:
+
+```[json]
+{
+  "aps.thread-id": "chat"
+}
+```
+
+
+Add FCM or APNs objects alongside the general notification and data fields to customize payloads for specific services:
+
+```[json]
+{
+  "notification": {
+    "title": "Hello from Ably!",
+    "body": "Example push notification from Ably."
+  },
+  "data": {
+    "foo": "bar",
+    "baz": "qux"
+  },
+  "apns": {
+    "aps": {
+      "alert": {
+        "title": "Hello to iOS from Ably!"
+      }
+    }
+  },
+  "fcm": {
+    "notification": {
+      "color": "#d3d3d3"
+    }
+  }
+}
+```
+
+This structure allows for targeted notifications with platform-specific settings and customization.
+
+
+h3(#apns-headers). APNs Headers
+
+To enable background notifications on iOS, you must include specific headers in the notification. You can specify these headers by adding an @apns-headers@ object, which should contain all the desired APNs headers, as part of the @apns@ object. The following example shows how a background notification would appear:
+
+```[json]
+{
+  "data": {
+    "foo": "bar",
+    "baz": "qux"
+  },
+  "apns": {
+    "aps": {
+      "content-available": 1,
+    },
+    "apns-headers": {
+      "apns-push-type": "background",
+      "apns-priority": "5"
+    }
+  }
+}
+```
+
+h2(#push-admin). Push admin API
```

Comment:
Do you think this content needs to be split across the sections and the usage highlighted for each? 

E.g. the difference between the 2 endpoints could be explained in the intro page.